docs/getting-started/products/yvaults/v3.md

```diff
@@ -0,0 +1,238 @@
+
+The vault will request back funds if the strategy has more debt than `max_debt`. These funds may be locked in the strategy, which would result in the strategy returning fewer funds than requested by the vault. 
+
+## Roles
+Vault functions that are permissioned will be callable by accounts with specific roles. 
+
+These roles are: 
+- `STRATEGY_MANAGER`: role for accounts that can add, remove, or migrate strategies.
+- `DEBT_MANAGER`: role for accounts that can rebalance and manage debt-related parameters.
+- `EMERGENCY_MANAGER`: role for accounts that can activate the shutdown mode.
+- `ACCOUNTING_MANAGER`: role for accounts that can process reports (save strategy p&l) and change accountant parameters.
+
+Every role can be filled by an externally owned account, multi-signature wallet, or other smart contracts. Each role can be filled by several accounts.
+
+The account that manages roles is a single account, set in `role_manager`.
+
+This `role_manager` can be an externally owned account, a multi-signature wallet, or a Governance Module that relays calls. 
+
+## Strategy Minimum API
+Strategies are completely independent smart contracts that can be implemented following the proposed template or in any other way.
+
+In any case, to be compatible with the vault, they need to implement the following functions, which are a subset of ERC4626 vaults: 
+- `asset()`: View the returning underlying asset.
+- `vault()`: View the returning vault this strategy is plugged into.
+- `totalAssets():` View the returning amount of assets. Which can include rewards valued in `asset`.
+- `maxDeposit(address)`: View the returning amount max that the strategy can take safely.
+- `deposit(assets, receiver)`: Deposits `assets` amount of tokens into the strategy. It can be restricted to the vault only or be open.
+- `maxWithdraw(address)`: View how many assets can be taken from the vault at any given point in time.
+- `withdraw(assets, receiver, owner)`: Withdraws `assets` amount of tokens from the strategy.
+- `balanceOf(address)`: View the number of shares of the strategy that the address has.
+
+This means that the vault can deposit into any ERC4626 vault and that a non-compliant strategy can be implemented provided that these functions have been implemented (even if they are in a non-ERC4626 compliant way). 
+
+Anything else is left to the strategy writer. However, to make security review easier, Yearn's template has the following optional functions: 
+- `tend()`: a function that will be called by bots. It will do anything to maintain a position, such as act on certain triggers.
+- `tendTrigger()`: implementation which triggers bots that will call tend function on the contract.
+- `invest()`: deposit funds into the underlying protocol.
+- `emergencyFreeFunds()`: close the position and return funds to the strategy. Losses can be accepted here.
+
+## ERC4626 Compliance
```

Comment:
This is referenced above in a hackmd I recommend moving above or deleting that section above.

Reply:
the hackmd was deleted and instead it points to this section now